src/Slick/Serve.hs

```diff
@@ -0,0 +1,98 @@
+{-# LANGUAGE OverloadedStrings #-}
+
+module Slick.Serve ( serverStart
+                   , serverStop
+                   , serverHandler
+                   ) where
+
+import           Control.Concurrent.MVar
+import           Control.Monad                  (forM_, forever, liftM, mzero,
+                                                 void, when, (>=>))
+import           Control.Monad.IO.Class
+import           Data.List
+import           Data.Monoid
+import           Data.String
+import qualified Data.Text                      as T
+import qualified Data.Text.Lazy                 as TL
+import           Network.HTTP.Types.Status      (Status)
+import qualified Network.Wai                    as Wai
+import qualified Network.Wai.Application.Static as Static
+import qualified Network.Wai.Handler.Warp       as Warp
+import qualified Network.Wai.Handler.WebSockets as Wai.WS
+import qualified Network.WebSockets             as WS
+
+--------------------------------------------------------------------------------
+
+type WSAppState = MVar [WS.Connection]
+
+-- | Run server with Websocket connection
+--   for live reloading
+serverStart :: FilePath                               -- ^ Directory to serve
+            -> String                                 -- ^ Host to bind on
+            -> Int                                    -- ^ Port to listen on
+            -> (WSAppState -> WS.Connection -> IO ()) -- ^ Websockets handler
+            -> IO ()
+serverStart directory host portWS wsApp = do
+  wsState <- newMVar []
+  Warp.runSettings
+    warpSettings $
+      Wai.WS.websocketsOr
+        WS.defaultConnectionOptions          -- Compression disabled
+        (WS.acceptRequest >=>                -- App for websocket requests
+          \conn -> do
```

Comment:
Why have a do block and then use `>>`? Mixing in `>=>` and then requiring a lambda instead of committing to `do` seems strange to.

I much prefer to just use `do` notation everywhere, it's easier to read and much more approachable for new folks who want to read the source.

You don't need to change all of this if you don't want to; but in the future I'd prefer to avoid `>=>` and usually even `>>` when possible.